Extract vue components to separate file and compile using vueify

deringer opened the pull request:
The first step of the VueJS refactor is to extract all of the components
into separate `*.vue` files and import them as components to the main
Vue instance - or, as in the case of FriendsList - into the Conference
component. This is just a placeholder PR to make sure I'm heading in
the right direction.

Once the underlying component structure is settled on, we can start
working on actually refactoring the code as needed.

- [x] Extract components to separate files for vueify
- [x] Establish an appropriate import style
- [x] Make sure that each component works as it did previously

mattstauffer commented on resources/assets/js/components/Conference.vue:
```diff
@@ -0,0 +1,31 @@
+        },
+
+        components: {
+            FriendsList: require('./FriendsList.vue')
```
i've been wondering how we should handle inline requiring as we move into ES6 style import. Because, right now, we're mixing them. But I don't know how you you should import a .vue component inside another using the ES6 notation. Let's keep this for now but hopefully we figure out what the actual best practice is there.

deringer replied:
@mattstauffer would it make sense to import the modules directly within the component itself? That way each component would explicitly define it's dependencies and we'd have the benefit of a consistent notation. 

```javascript
<script>
    import FriendsList from './FriendsList.vue';

    export default {
        components: { FriendsList }
    };
</script>
```

mattstauffer replied:
i like that :)

mattstauffer commented on resources/assets/js/app.js:
```diff
@@ -13,256 +13,20 @@ var setErrorsOnForm = function (form, errors) {
-Vue.component('form-errors', {
-    props: ['form'],
+import Dashboard from './components/Dashboard.vue';
+import FriendsList from './components/FriendsList.vue';
```
I think this line is unnecessary now?

mattstauffer commented on resources/assets/js/components/Conference.vue:
```diff
@@ -0,0 +1,31 @@
+            FriendsList: require('./FriendsList.vue')
+        },
+
+        ready: function () {
```
No need to keep any properties (`ready` and `methods`) we're not using :)

mattstauffer commented on resources/assets/js/components/Dashboard.vue:
```diff
@@ -0,0 +1,152 @@
+    export default {
+        data: function() {
+            return {
+                currentUserId: Confomo.userId,
```
Wondering whether this was the property you were asking about getting. :) It's better to get it from a global variable like this than to get it from the DOM, but I'm betting it's still better if we do it a more Vue-y way like passing it in as a property. I might play around with a few options.

deringer replied:
It was actually a work thing: I was probably thinking about it the wrong way around. In my blade template I had a section that I was trying to bind stuff into, which I thought would work based on the fact the section was inside the element Vue was bound to.

```
@section('page_title', 'Items <span v-show="items.length > 0">(@{{ items.length }})</span>')
```

Vue kept complaining that it couldn't parse the `v-show` expression (or similar). I suspect that this is due to me trying to access a component-scoped variable outside of the component. I tweaked the interface in that instance to render the number of items inside the component in the end :+1:

mattstauffer commented on resources/assets/js/components/Dashboard.vue:
```diff
@@ -0,0 +1,152 @@
+                        this.addConferenceForm.name = '';
+                        this.addConferenceForm.adding = false;
+                        // @todo: Add new conference to the list more cleanly
+                        this.getAllConferences();
```
Ha, yah, I really didn't know what I was doing. We can just push the conference onto this.conferences; we don't have to re-get from the API.

deringer commented:
@mattstauffer I think we're in a happy place with this separation of modules now. Haven't had a chance to work on ConFOMO directly for a couple of weeks, but I've been playing with Vue on some work stuff, so working with it makes sense for me now.

I finished up addressing your feedback, so if you can review when you get a chance and merge, I'll continue working. On that note, would you prefer that I continue with feature branches off of the `complete-refactor` branch, or just continue working off that branch with future changes?

mattstauffer commented:
I want to publicly acknowledge that I'm a big jerk for waiting so long to review this. :(

mattstauffer commented:
<3

deringer commented:
There's a whole new major release of Elixir since I made this PR 😂
